RockWeb/Webhooks/WebhookToWorkflow.ashx

```diff
@@ -0,0 +1,394 @@
+    protected DefinedValueCache GetHookForRequest( Guid definedTypeGuid )
+    {
+        var hooks = DefinedTypeCache.Read( definedTypeGuid );
+        if ( hooks != null )
+        {
+
+            foreach ( DefinedValueCache hook in hooks.DefinedValues.OrderBy( h => h.Order ) )
+            {
+                if ( IsHookValidForRequest( hook ) )
+                {
+                    return hook;
+                }
+            }
+        }
+        return null;
+    }
+
+    /// <summary>
+    /// Check if this DefinedValue is valid for the current request. Subclasses should
+    /// call the base method so that the Method and Url are verified.
+    /// </summary>
+    /// <param name="hook">The DefinedValue that is to be considered for this request.</param>
+    /// <returns>true if the DefinedValue matches this request, otherwise false.</returns>
+    protected virtual bool IsHookValidForRequest( DefinedValueCache hook )
+    {
+        string hookUrl = hook.GetAttributeValue( "Url" );
+        string hookMethod = hook.GetAttributeValue( "Method" );
+
+        //
+        // Check for match on method type, if not continue to the next item.
+        //
+        if ( !string.IsNullOrEmpty( hookMethod ) && !HttpContext.Request.HttpMethod.ToString().Equals( hookMethod, StringComparison.InvariantCultureIgnoreCase ) )
+        {
+            return false;
+        }
+
+        //
+        // Check for match on the URL.
+        //
+        if ( string.IsNullOrEmpty( hookUrl ) )
+        {
+            return true;
+        }
+        else if ( hookUrl.StartsWith( "^" ) && hookUrl.EndsWith( "$" ) )
+        {
+            return Regex.IsMatch( Url, hookUrl, RegexOptions.IgnoreCase );
+        }
+        else
+        {
+            return Url.Equals( hookUrl, StringComparison.InvariantCultureIgnoreCase );
+        }
+    }
+
+    /// <summary>
+    /// Populates any defined Workflow attributes specified to this webhook type.
+    /// Subclasses may call the base method to have the Request attribute set.
+    /// </summary>
+    /// <param name="workflow">The workflow whose attributes need to be set.</param>
+    /// <param name="hook">The DefinedValue of the currently executing webhook.</param>
+    protected virtual void PopulateWorkflowAttributes( Workflow workflow, DefinedValueCache hook )
+    {
+        workflow.SetAttributeValue( "Request", JsonConvert.SerializeObject( RequestToDictionary( hook ) ) );
+
+        string workflowAttributes = hook.GetAttributeValue( "WorkflowAttributes" );
+        string nameTemplate = hook.GetAttributeValue( "WorkflowNameTemplate" );
+
+        string[] attributes = workflowAttributes.Split( '|' );
+
+        foreach ( string attribute in attributes )
+        {
+            if ( attribute.Contains( '^' ) )
+            {
+                string[] settings = attribute.Split( '^' );
+                workflow.SetAttributeValue( settings[0], settings[1] );
```

Comment:
The "Value" part of the KeyValues (settings[1] will be Lava, so it needs to be resolved with the mergeFields dictionary...
```
workflow.SetAttributeValue( settings[0], settings[1].ResolveMergeFields( mergeFields ) );
```